oap-server/server-starter/src/main/java/org/apache/skywalking/oap/server/starter/config/ApplicationConfigLoader.java

```diff
@@ -61,21 +62,25 @@ private void loadConfig(ApplicationConfiguration configuration) throws ConfigFil
                     if (providerConfig.size() > 0) {
                         logger.info("Get a module define from application.yml, module name: {}", moduleName);
                         ApplicationConfiguration.ModuleConfiguration moduleConfiguration = configuration.addModule(moduleName);
-                        providerConfig.forEach((name, propertiesConfig) -> {
-                            logger.info("Get a provider define belong to {} module, provider name: {}", moduleName, name);
+                        providerConfig.forEach((providerName, propertiesConfig) -> {
+                            logger.info("Get a provider define belong to {} module, provider name: {}", moduleName, providerName);
                             Properties properties = new Properties();
                             if (propertiesConfig != null) {
-                                propertiesConfig.forEach((key, value) -> {
-                                    properties.put(key, value);
-                                    final Object replaceValue = yaml.load(PropertyPlaceholderHelper.INSTANCE
-                                        .replacePlaceholders(value + "", properties));
-                                    if (replaceValue != null) {
-                                        properties.replace(key, replaceValue);
+                                propertiesConfig.forEach((propertyName, propertyValue) -> {
+                                    if (propertyValue instanceof Map) {
```

Comment:
I think Map and Properties are good choices for here.

Reply:
And is this covered by unit test?

Reply:
Hi, where should i add unit test?

Reply:
What do you mean where? Same module, test folder, same package, ClassName + Test. As usual. We don't have a special rule for this.

Reply:
You mean create a new unit test class, and write unit class for ...loadConfig()?
@wu-sheng

Reply:
ApplicationConfigLoader doesn't have Unit Test today, please add one.

Reply:
Finished it.